plasmapy/formulary/parameters.py

```diff
@@ -669,13 +670,22 @@ def thermal_speed(
     try:
         coef = _coefficients[ndim]
     except KeyError:
-        raise ValueError("{ndim} is not a supported value for ndim in thermal_speed")
+        raise ValueError(
+            f"Got unsupported value for argument 'ndim' ({ndim}), expect one of "
+            f"{list(_coefficients.keys())}."
+        )
     try:
         coef = coef[method]
     except KeyError:
-        raise ValueError("Method {method} not supported in thermal_speed")
+        raise ValueError(
+            f"Got unsupported value for argument 'method' ({method}), expected one "
+            f"of {list(coef.keys())}."
+        )
+
+    speed = (k_B * T / m).value
+    speed = np.sqrt(coef * speed) * u.m / u.s
```

Comment:
I don't think this will work since you still need to set `unit = u.m/u.s` within `thermal_speed` and that is not being timed in your last two cases.  So, it looks faster since you're not timing something that still need to be apart of `thermal_speed`.

A work around, we could create a utility package, e.g. `plasmapy.utils.unit_defaults`, where we define all of our default units.   For example,

```python
>>> import plasmapy.utils.unit_defaults as ud
>>> ud.speed == u.m / u.s
True
```

Reply:
I think @StanczakDominik was suggesting that you could define `unit` outside of the function (the way some of the constants are defined)? Then it would be evaluated once when the module is imported?